### PR TITLE
Adding Channels and Orchestration

### PR DESCRIPTION
**Pull request by ameyc**

### CODE REVIEW

File: Cargo.toml [hunk 1]
```diff
@@ -18,8 +18,9 @@ description = "Embeddable stream processing engine"
 [workspace.dependencies]
 denormalized = { path = "crates/core" }
 denormalized-common = { path = "crates/common" }
+denormalized-orchestrator = { path = "crates/orchestrator" }
 
-datafusion = { git = "https://github.com/probably-nothing-labs/arrow-datafusion", branch = "main" }
+datafusion = { git = "https://github.com/probably-nothing-labs/arrow-datafusion", branch = "amey/patch-with-node-id" }
```
Comment (emgeee) [inline Cargo.toml:23]: IMO we want to keep the node-id changes, we should merge them into https://github.com/probably-nothing-labs/arrow-datafusion main branch

File: crates/core/src/physical_plan/stream_table.rs [hunk 2]
```diff
@@ -0,0 +1,325 @@
...
+
+    fn as_partition_with_node_id(&self) -> Option<&KafkaStreamRead> {
+        None
+    }
+}
+
+/// An [`ExecutionPlan`] for one or more [`PartitionStream`]s.
+///
+/// If your source can be represented as one or more [`PartitionStream`]s, you can
+/// use this struct to implement [`ExecutionPlan`].
```
Comment (emgeee) [inline crates/core/src/physical_plan/stream_table.rs:70]: Should add a comment/description as to why DenormalizedStreamingTableExec was created

File: crates/orchestrator/Cargo.toml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+[package]
+name = "denormalized-orchestrator"
+version = { workspace = true }
+edition = { workspace = true }
+
+[dependencies]
+anyhow = "1.0.86"
```
Comment (emgeee) [inline crates/orchestrator/Cargo.toml:7]: is this dependency needed?

File: Cargo.toml (hunk 1 above)
Comment (ameyc) [reply Cargo.toml:23]: +1. makes sense to merge that one after the AggregateExprs PR.

**Review by emgeee — APPROVED** [review]

**Pull request merged**